File: solutions/java/allergies/1/src/main/java/Allergies.java
import java.util.Arrays;
import java.util.List;

class Allergies {
  private final int score;

  Allergies(int score) {
    this.score = score;
  }

  boolean isAllergicTo(Allergen allergen) {
    return (score & allergen.getScore()) > 0;
  }

  List<Allergen> getList() {
    return Arrays.stream(Allergen.values()).filter(this::isAllergicTo).toList();
  }
}

File: solutions/kotlin/allergies/1/src/main/kotlin/Allergies.kt
class Allergies(score: Int) {
    private val allergens = Allergen.entries.filter { score and it.score != 0 }

    fun getList(): List<Allergen> = allergens

    fun isAllergicTo(allergen: Allergen): Boolean = allergens.contains(allergen)
}
